src/utils/PageParsing.ts

```diff
@@ -18,13 +18,73 @@ import {PackageURL} from 'packageurl-js';
 import {FORMATS, RepoType} from './Constants';
 
 const getArtifactDetailsFromDOM = (repoFormat: RepoType, url: string): PackageURL | undefined => {
-  console.info('url', url);
-  console.info('format', repoFormat.repoFormat);
+  console.info('getArtifactDetailsFromDOM url', url, repoFormat.repoFormat);
 
   if (repoFormat.repoFormat === FORMATS.npm) {
     return parseNPM(url);
   } else if (repoFormat.repoFormat === FORMATS.nuget) {
     return parseNuget(url);
+  } else if (repoFormat.repoFormat === FORMATS.golang) {
+    // console.log('parsegolang', url);
+    const purl: PackageURL | undefined = parseGolang(url);
+
+    // console.log('purl', purl);
+    return purl;
+  }
+
+  return undefined;
+};
+
+const parseGolang = (url: string): PackageURL | undefined => {
+  console.log('parseGolang', url);
+  // const name = 'etcd-io/etcd';
+  // const version = 'v3.3.25+incompatible';
+  // pkg:GOLANG/google.golang.org/genproto#/googleapis/api/annotations/;
+  // return generatePackageURL(FORMATS.golang, packageId, version);
+  // The following coordinates are missing for given format: [version]
+  //https://pkg.go.dev/github.com/etcd-io/etcd@v0.3.0 ->CVE-2020-15115, CVE - 2020 - 15136;
+  //https://pkg.go.dev/github.com/etcd-io/etcd -> v->v3.3.25+incompatible ->unknown
+  //https://pkg.go.dev/github.com/go-gitea/gitea ->Version: v1.8.3 ->CVE-2018-15192 and others
+
+  const elements = url.split('/');
+  const type = 'golang';
+  const namespace = 'github.com';
```

Comment:
This will not always be the case. You'll see golang.org, etc... and a bunch of others over time, especially golang.org